public/app/core/components/TransformersUI/FilterByValueTransformerEditor.tsx

```diff
@@ -0,0 +1,263 @@
+            { label: 'Exclude', value: 'exclude' },
+          ]}
+          value={config.type}
+          onChange={option => {
+            // console.log('onChange filterType', option.value);
+            onConfigChange({ ...config, type: option.value || 'include' });
+          }}
+          menuPlacement="bottom"
+        />
+      </div>
+      <div className="gf-form gf-form-spacing">
+        <div className="gf-form-label width-8">Rows with</div>
+        <Select
+          className="width-16"
+          placeholder="Field Name"
+          options={fieldNameOptions}
+          value={config.fieldName}
+          invalid={fieldNameInvalid}
+          onChange={value => {
+            onConfigChange({
+              ...config,
+              fieldName: value?.value ?? null,
+            });
+          }}
+          isClearable
+          menuPlacement="bottom"
+        />
+      </div>
+      <div className="gf-form gf-form-spacing">
+        <div className="gf-form-label width-8">Matching</div>
+        <Select
+          invalid={filterTypeInvalid}
+          className="width-8"
+          placeholder="Select test"
+          options={filterTypeOptions}
+          value={config.filterType}
+          onChange={value => {
+            onConfigChange({ ...config, filterType: (value.value as ValueFilterID) ?? ValueFilterID.regex });
+          }}
+          menuPlacement="bottom"
+        />
+      </div>
+      <div className="gf-form gf-form--grow gf-form-spacing ">
+        {filterInfo.placeholder && (
+          <Input
+            className="flex-grow-1"
+            invalid={filterExpressionInvalid}
+            defaultValue={config.filterExpression || undefined}
+            placeholder={filterInfo.placeholder}
+            onBlur={event => {
+              onConfigChange({ ...config, filterExpression: event.currentTarget.value });
+            }}
+          />
+        )}
+      </div>
+      <div className="gf-form">
+        <Button icon="times" onClick={onDelete} style={{ height: '100%' }} size="sm" variant="secondary" />
+      </div>
+    </div>
+  );
+};
+
+export const FilterByValueTransformerEditor: React.FC<TransformerUIProps<FilterByValueTransformerOptions>> = ({
+  input,
+  options,
+  onChange,
+}) => {
+  const fieldsInfo = useMemo(() => getAllFieldInfoFromDataFrames(input), [input]);
+  const fieldNameOptions = useMemo(
+    () =>
+      fieldsInfo
+        .filter((item: Record<string, any>) => item.type !== FieldType.time)
+        .map((item: Record<string, any>) => ({ label: item.name, value: item.name })),
+    [fieldsInfo]
+  );
+
+  const onAddFilter = useCallback(() => {
+    let valueFilters = options.valueFilters.map(filter => ({ ...filter })); // Deep copy
+    valueFilters.push({
+      type: 'include',
+      fieldName: null,
+      filterExpression: null,
+      filterType: ValueFilterID.regex,
+    });
+
+    onChange({
+      ...options,
+      valueFilters,
+    });
+  }, [options]);
```

Comment:
```suggestion
  }, [onChange , options]);
```
All the variables (mutable) that are from the outer scope should be part of the dependency list.